src/main/java/org/springframework/integration/splunk/support/SplunkHECWriter.java

```diff
@@ -0,0 +1,459 @@
+
+	private CloseableHttpAsyncClient httpClient;
+	private URI uri;
+
+	private static final HostnameVerifier HOSTNAME_VERIFIER = new HostnameVerifier() {
+		public boolean verify(String s, SSLSession sslSession) {
+			return true;
+		}
+	};
+
+	public SplunkHECWriter(ServiceFactory serviceFactory, Args args) {
+
+		this.serviceFactory = serviceFactory;
+		this.args = args;
+	}
+
+	// getters and setters for various configurable parameters
+
+	public String getToken() {
+		return token;
+	}
+
+	public String getHost() {
+		return host;
+	}
+
+	public int getPort() {
+		return port;
+	}
+
+	public boolean isHttps() {
+		return https;
+	}
+
+	public int getPoolsize() {
+		return poolsize;
+	}
+
+	public String getIndex() {
+		return index;
+	}
+
+	public String getSource() {
+		return source;
+	}
+
+	public String getSourcetype() {
+		return sourcetype;
+	}
+
+	public boolean isBatchMode() {
+		return batchMode;
+	}
+
+	public long getMaxBatchSizeBytes() {
+		return maxBatchSizeBytes;
+	}
+
+	public long getMaxBatchSizeEvents() {
+		return maxBatchSizeEvents;
+	}
+
+	public long getMaxInactiveTimeBeforeBatchFlush() {
+		return maxInactiveTimeBeforeBatchFlush;
+	}
+
+	public void setToken(String token) {
+		this.token = token;
+	}
+
+	public void setHost(String host) {
+		this.host = host;
+	}
+
+	public void setPort(int port) {
+		this.port = port;
+	}
+
+	public void setHttps(boolean https) {
+		this.https = https;
+	}
+
+	public void setPoolsize(int poolsize) {
+		this.poolsize = poolsize;
+	}
+
+	public void setIndex(String index) {
+		this.index = index;
+	}
+
+	public void setSource(String source) {
+		this.source = source;
+	}
+
+	public void setSourcetype(String sourcetype) {
+		this.sourcetype = sourcetype;
+	}
+
+	public void setBatchMode(boolean batchMode) {
+		this.batchMode = batchMode;
+	}
+
+	public void setMaxBatchSizeBytes(long maxBatchSizeBytes) {
+		this.maxBatchSizeBytes = maxBatchSizeBytes;
+	}
+
+	public void setMaxBatchSizeEvents(long maxBatchSizeEvents) {
+		this.maxBatchSizeEvents = maxBatchSizeEvents;
+	}
+
+	public void setMaxInactiveTimeBeforeBatchFlush(
+			long maxInactiveTimeBeforeBatchFlush) {
+		this.maxInactiveTimeBeforeBatchFlush = maxInactiveTimeBeforeBatchFlush;
+	}
+
+	@Override
+	public void write(SplunkEvent data) throws Exception {
+
+		String message = data.toString();
+
+		String currentMessage = "";
+		try {
+
+			if (!(message.startsWith("{") && message.endsWith("}"))
+					&& !(message.startsWith("\"") && message.endsWith("\"")))
+				message = wrapMessageInQuotes(message);
+
+			// could use a JSON Object , but the JSON is so trivial , just
+			// building it with a StringBuffer
+			StringBuffer json = new StringBuffer();
+			json.append("{\"").append("event\":").append(message).append(",\"")
+					.append("index\":\"").append(getIndex()).append("\",\"")
+					.append("source\":\"").append(getSource()).append("\",\"")
+					.append("sourcetype\":\"").append(getSourcetype())
```

Comment:
Looks like these options are redundant and we really can use those from the root components and just play here a bit with the `Args` like it is done in other writers.